CONTRIBUTING.md

```diff
@@ -94,137 +65,217 @@ script before each commit and abort the commit on error. To
+- Some commands, such as `dotnet build` and `npm run build:d` may need `sudo` prefixed to execute successfully.
 
-- Run `npm run start:dev` to start the webpack-dev-server
+### Github Codespaces
 
-- Start debugger (making sure the webpack-dev-server is still running). The default launch configuration should be set to Launch Firefox.
+1. If necessary, read Github's [codespace creation](https://docs.github.com/en/codespaces/developing-in-codespaces/creating-a-codespace) documentation.
 
-- Once Firefox starts up, you should get an untrusted certificate error page. Select Advanced and then select **Accept the Risk and Continue** and log into your Azure DevOps account. From now on, if you leave this Firefox window open, the debugger will reattach instead of starting a clean Firefox instance each time.
+2. Create a branch for the work you are planning to complete.
 
-- Once you are logged in to Azure DevOps, your extension should be running. Set a breakpoint in a method in VS Code and you should see that breakpoint hit when that method executes.
+3. Create a codespace through the new branch's dropdown on Github.
 
-### Storage
+4. Codespaces are powered by [Visual Studio Code and dev containers](https://docs.github.com/en/codespaces/developing-in-codespaces/developing-in-a-codespace) - once the codespace has been entered, you can follow the instructions within the [dev containers section](#dev-containers) for continuing to set up the development environment. View the [Developing in a codespace](https://docs.github.com/en/codespaces/developing-in-codespaces/developing-in-a-codespace) documentation for additional details.
 
-The Retrospectives tool uses the [Azure DevOps data service](https://docs.microsoft.com/en-us/azure/devops/extend/develop/data-storage?view=vsts) for handling all its storage.
+## Build
+
+### Test in Azure DevOps
+
+Test changes in the Azure DevOps environment by publishing a development version of the extension under an Azure DevOps publisher account.
+
+---
+
+1. Clone this repository, and open in your preferred [development environment](#development-environments).
+2. Using Powershell, navigate to the '/RetrospectiveExtension.Frontend' folder, run `npm install`. This will download all the dependent packages listed in 'package.json'.
+3. Copy the file `RetrospectiveExtension.Frontend\config\environment.tsx.template` into `RetrospectiveExtension.Frontend\config\environment.tsx` and update the fields
+
+    ```json
+    {
+      CollaborationStateServiceUrl : "https://my-backend-service.com", // change this to the deployed backend service
+      AppInsightsInstrumentK       : "my_instrumentation_key" // put Instrumentation key here
+    }
+    ```
+
+4. Run `npm run build:d` to build the project. Refer to the 'scripts' section in 'package.json' for other commands.
```

Comment:
Would be great to mention that `:d` is for development and `:p` is for production build which has smaller bundle size.

Reply:
ah - yes I was thinking about this but was unsure if we want to encourage people to use `:p`. I will add a note

Reply:
Yes, testing the production build before creating/merging PR is good to make sure that the build doesn't break in production environment, which can be more expensive to fix.